go/tasks/plugins/array/k8s/task.go

```diff
@@ -0,0 +1,232 @@
+package k8s
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	idlCore "github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/utils"
+	"github.com/lyft/flyteplugins/go/tasks/plugins/array"
+	"github.com/lyft/flyteplugins/go/tasks/plugins/array/arraystatus"
+	arrayCore "github.com/lyft/flyteplugins/go/tasks/plugins/array/core"
+	"github.com/lyft/flyteplugins/go/tasks/plugins/array/errorcollector"
+	"github.com/lyft/flytestdlib/bitarray"
+	errors2 "github.com/lyft/flytestdlib/errors"
+	"github.com/lyft/flytestdlib/logger"
+	"github.com/lyft/flytestdlib/storage"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sTypes "k8s.io/apimachinery/pkg/types"
+)
+
+type Task struct {
+	LogLinks         []*idlCore.TaskLog
+	State            *arrayCore.State
+	NewArrayStatus   *arraystatus.ArrayStatus
+	Config           *Config
+	ChildIdx         int
+	MessageCollector *errorcollector.ErrorMessageCollector
+}
+
+type TaskStatus int8
+
+const (
+	Success TaskStatus = iota
+	Error
+	Waiting
+	ReturnState
+)
+
+func (t Task) Launch(ctx context.Context, tCtx core.TaskExecutionContext, kubeClient core.KubeClient) (TaskStatus, error) {
+	podTemplate, _, err := FlyteArrayJobToK8sPodTemplate(ctx, tCtx)
+	if err != nil {
+		return Error, errors2.Wrapf(ErrBuildPodTemplate, err, "Failed to convert task template to a pod template for a task")
+	}
+
+	var args []string
+	if len(podTemplate.Spec.Containers) > 0 {
+		args = append(podTemplate.Spec.Containers[0].Command, podTemplate.Spec.Containers[0].Args...)
```

Comment:
do we always assume that there is at most one container in the podspec? should we error if there are more? 0? in line 60 and elsewhere you reference the Containers[0] again without this check

Reply:
I'm not entirely sure. This was the code that was there before hand.

Reply:
hm it seems odd that we check before de-referencing here but now below, can we maybe check and just throw an error?

Reply:
Updated to throw error. Not sure if this is what you meant or throw error below?

Reply:
@katrogan ^^

Reply:
yes, thank you!